### PR TITLE
Implement tagEvent (Part 4)

### PR DESCRIPTION
**Pull request by johnxnguyen**
## What's new in this PR?

This is a part of an ongoing effort to refactor the analytics system to align it to the implementation spec and improving and cleaning up the code along the way.

### Changes

- Add attribute keys for `appName` and `appVersion`
- Implement `tagEvent`
- Remove old protocol no longer in use

### Note

Tests are failing because I removed the old `tagEvent` implementation. I'll solve this in upcoming PRs.

### CODE REVIEW

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -6461,6 +6454,16 @@
 			path = Countly;
 			sourceTree = "<group>";
 		};
+		EE430EBA25C208A90056BDE3 /* Screen Curtain */ = {
+			isa = PBXGroup;
+			children = (
+				BFBBD1EC20D9238B00BFC4AD /* ScreenCurtain.swift */,
+				EE430EC725C20D080056BDE3 /* ScreenCurtainDelegate.swift */,
+				8FC85255199245760008B66B /* ScreenCurtain.RootViewController.swift */,
+			);
```
Comment (KaterinaWire) [inline Wire-iOS.xcodeproj/project.pbxproj:6463]: Are these changes necessary?

Comment (johnxnguyen) [reply Wire-iOS.xcodeproj/project.pbxproj:6463]: These weren't added, probably Xcode just revalidated its project file.

**Review by KaterinaWire — APPROVED** [review]

**Review by David-Henner — APPROVED** [review]
Just pointing out that there are some TODO comments left in the code, but assuming it is intentional
